File: Prototype_Apache2/src/java/aa/Ask.java
package aa;

import java.util.Date;

// represents an Ask (in a sell order)
public class Ask {
  private int askID;
  private String stock;
  private int price; // ask price
  private String userId; // user who made this sell order
  private Date date;
  
  public Ask(int id, String stock,int price,String userId,Date date){
      this.askID = id;
      this.stock = stock;
      this.price = price;
      this.userId = userId;
      this.date = date;
  }
  
  // constructor
  public Ask(String stock, int price, String userId) {
    this.stock = stock;
    this.price = price;
    this.userId = userId;
    this.date = new Date();
  }
  
  public Ask(String stock,int price,String userId,Date date){
      this.stock = stock;
      this.price = price;
      this.userId = userId;
      this.date = date;
  }
  
  // getters
  public String getStock() {
    return stock;
  }

  public int getPrice() {
    return price;
  }

  public String getUserId() {
    return userId;
  }

  public Date getDate() {
    return date;
  }
  
  public int getID(){
      return askID;
  }
  
  public void setID(int i){
      this.askID = i;
  }

  // toString
  public String toString() {
    return "id: " + askID + ", stock: " + stock + ", price: " + price + ", userId: " + userId + ", date: " + date;
  }
}

File: AA_Domain/routes/modules/ask.js
function Ask(idvalue, stock, price, userId, date) {
	this.idvalue = idvalue;
	this.stock = stock;
	this.price = price;
	this.userId = userId;
	this.date = date;
}

Ask.prototype.getStock = function() {
	return this.stock;
}

Ask.prototype.getPrice = function() {
	return this.price;
}

Ask.prototype.getUserId = function() {
	return this.userId;
}

Ask.prototype.getDate = function() {
	return this.date;
}

Ask.prototype.getId = function() {
	return this.idvalue;
}

Ask.prototype.setId = function(idvalue) {
	this.idvalue = idvalue;
}

Ask.prototype.getDateString = function() {
	var dateString = [[AddZero(this.date.getDate()), AddZero(this.date.getMonth() + 1), this.date.getFullYear()].join("/"), [AddZero(this.date.getHours()), AddZero(this.date.getMinutes())].join(":"), this.date.getHours() >= 12 ? "PM" : "AM"].join(" ");
	return dateString;
}

Ask.prototype.toString = function() {
	var dateString = [[AddZero(this.date.getDate()), AddZero(this.date.getMonth() + 1), this.date.getFullYear()].join("/"), [AddZero(this.date.getHours()), AddZero(this.date.getMinutes())].join(":"), this.date.getHours() >= 12 ? "PM" : "AM"].join(" ");
	return "stock: " + this.stock + ", price: " + this.price + ", userId: " + this.userId + ", date: " + dateString;
}

//Pad given value to the left with "0"
function AddZero(num) {
    return (num >= 0 && num < 10) ? "0" + num : num + "";
}

module.exports.Ask = Ask;